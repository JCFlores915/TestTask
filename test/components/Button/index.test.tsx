import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CustomButton from '../../../src/components/Button';
import React from 'react';

describe('Button Component', () => {
    test('Renderizar el CustomButton  de manera correcta', () => {
        const wrapper = renderer.create(<CustomButton type="warning" text="Home"  />).toJSON();
        expect(wrapper).toMatchSnapshot();        
    });

    test('Renderizar el CustomButton de manera correcta con block vacío', () => {
        const wrapper = renderer.create(<CustomButton type="warning" text="Home" block="" />).toJSON();
        expect(wrapper).toMatchSnapshot();        
    })

    test('Renderizar el CustomButton de manera correcta con diferentes tipos', () => {
        const wrapper = renderer.create(<CustomButton type="success" text="Submit" htmlType="submit" block="full" />).toJSON();
        expect(wrapper).toMatchSnapshot();        
    })

    test('Renderizar el CustomButton de manera correcta con el prop de texto vacío', () => {
        const wrapper = renderer.create(<CustomButton type="warning" text="" block="" />).toJSON();
        expect(wrapper).toMatchSnapshot();        
    })

    test(
        'Hacer click en el CustomButton y que se ejecute la funcion onClick'
        , () => {
        const onClickMock = jest.fn();
        const { getByText } = render(<CustomButton type="warning" text="Home" onClick={onClickMock} />);
        fireEvent.click(getByText('Home'));
        expect(onClickMock).toHaveBeenCalled();
    })

    
});
