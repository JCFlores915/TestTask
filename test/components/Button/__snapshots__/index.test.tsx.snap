// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renderizar el CustomButton  de manera correcta 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-zl9ks2 ant-btn-default btn-custom-warning"
  disabled={false}
  onClick={[Function]}
  style={{}}
  type="button"
>
  <span>
    Home
  </span>
</button>
`;

exports[`Button Component Renderizar el CustomButton de manera correcta con block vacío 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-zl9ks2 ant-btn-default btn-custom-warning"
  disabled={false}
  onClick={[Function]}
  style={{}}
  type="button"
>
  <span>
    Home
  </span>
</button>
`;

exports[`Button Component Renderizar el CustomButton de manera correcta con diferentes tipos 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-zl9ks2 ant-btn-default btn-custom-success-full"
  disabled={false}
  onClick={[Function]}
  style={{}}
  type="submit"
>
  <span>
    Submit
  </span>
</button>
`;

exports[`Button Component Renderizar el CustomButton de manera correcta con el prop de texto vacío 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-zl9ks2 ant-btn-default btn-custom-warning"
  disabled={false}
  onClick={[Function]}
  style={{}}
  type="button"
/>
`;
